src/scripts/librarymenu.js

```diff
@@ -155,297 +302,669 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
-        elem && (show ? elem.classList.remove("hide") : elem.classList.add("hide"))
+
+        if (elem) {
+            if (show) {
+                elem.classList.remove("hide");
+            } else {
+                elem.classList.add("hide");
+            }
+        }
     }
 
     function updateLibraryMenu(user) {
-        if (!user) return showBySelector(".libraryMenuDownloads", !1), showBySelector(".lnkSyncToOtherDevices", !1), void showBySelector(".userMenuOptions", !1);
-        user.Policy.EnableContentDownloading ? showBySelector(".lnkSyncToOtherDevices", !0) : showBySelector(".lnkSyncToOtherDevices", !1), user.Policy.EnableContentDownloading && appHost.supports("sync") ? showBySelector(".libraryMenuDownloads", !0) : showBySelector(".libraryMenuDownloads", !1);
-        var userId = Dashboard.getCurrentUserId(),
-            apiClient = getCurrentApiClient(),
-            libraryMenuOptions = document.querySelector(".libraryMenuOptions");
-        libraryMenuOptions && getUserViews(apiClient, userId).then(function(result) {
-            var items = result,
-                html = "";
-            html += '<h3 class="sidebarHeader">', html += globalize.translate("HeaderMedia"), html += "</h3>", html += items.map(function(i) {
-                var icon = "folder",
-                    itemId = i.Id;
-                "channels" == i.CollectionType ? itemId = "channels" : "livetv" == i.CollectionType && (itemId = "livetv"), "photos" == i.CollectionType ? icon = "photo_library" : "music" == i.CollectionType || "musicvideos" == i.CollectionType ? icon = "library_music" : "books" == i.CollectionType ? icon = "library_books" : "playlists" == i.CollectionType ? icon = "view_list" : "games" == i.CollectionType ? icon = "games" : "movies" == i.CollectionType ? icon = "video_library" : "channels" == i.CollectionType || "Channel" == i.Type ? icon = "videocam" : "tvshows" == i.CollectionType ? icon = "tv" : "livetv" == i.CollectionType && (icon = "live_tv"), icon = i.icon || icon;
-                i.onclick && i.onclick;
-                return '<a is="emby-linkbutton" data-itemid="' + itemId + '" class="lnkMediaFolder navMenuOption" href="' + getItemHref(i, i.CollectionType) + '"><i class="md-icon navMenuOptionIcon">' + icon + '</i><span class="sectionName navMenuOptionText">' + i.Name + "</span></a>"
-            }).join(""), libraryMenuOptions.innerHTML = html;
-            for (var elem = libraryMenuOptions, sidebarLinks = elem.querySelectorAll(".navMenuOption"), i = 0, length = sidebarLinks.length; i < length; i++) sidebarLinks[i].removeEventListener("click", onSidebarLinkClick), sidebarLinks[i].addEventListener("click", onSidebarLinkClick)
-        })
+        if (!user) {
+            showBySelector(".libraryMenuDownloads", false);
+            showBySelector(".lnkSyncToOtherDevices", false);
+            return void showBySelector(".userMenuOptions", false);
+        }
+
+        if (user.Policy.EnableContentDownloading) {
```

Comment:
this could probably be replaced with
```js
showBySelector(".lnkSyncToOtherDevices", !!user.Policy.EnableContentDownloading);
```

Reply:
Possibly. Commented a FIXME and your suggestion. Personally unsure if its equivalent.